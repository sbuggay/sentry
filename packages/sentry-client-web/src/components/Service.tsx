import * as React from "react";

import Status from "./Status";
import { EStatus } from "../constants";

import { IService } from "../reducer";

interface IStateProps {
    data: IService;
}

export class Service extends React.Component<IStateProps, any> {

    public getStyle(): React.CSSProperties {
        return {
            width: "280px",
            padding: "5px 10px",
            borderBottom: "1px solid #adb0af"
        };
    }

    // Render
    public renderStatus(): JSX.Element {
        const status = this.props.data.status ? EStatus.available : EStatus.outage;
        return (
            <Status status={status} />
        );
    }

    public render(): JSX.Element {
        return (
            <div
                style={this.getStyle()}>
                <div>
                    <span>
                        {this.renderStatus()}
                        {this.props.data.name}
                    </span>
                </div>
            </div>
        );
    }
}

export default Service;
