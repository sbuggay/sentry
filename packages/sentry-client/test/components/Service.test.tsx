import * as React from "react";
import * as renderer from "react-test-renderer";

import Service from "../../src/components/Service";

describe("Service", () => {

    it("renders a basic Service", () => {
        const component = renderer.create(
            <Service service={{name: "test", script: "test", test: "test", status: false}} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
