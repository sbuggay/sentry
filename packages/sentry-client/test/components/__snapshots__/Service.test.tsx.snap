// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service renders a basic Service 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid #adb0af",
      "padding": "5px 10px",
      "width": "280px",
    }
  }
>
  <div>
    <span>
      <i
        aria-hidden="true"
        className="fa fa-exclamation-triangle"
        style={
          Object {
            "color": "#DF3E2D",
            "marginRight": "0.5em",
          }
        }
      />
      test
    </span>
  </div>
</div>
`;
